backend/Tournament/views.py

```diff
@@ -0,0 +1,131 @@
+from .serializers import TournamentSerializer, \
+	 						TournamentCreationSerializer, \
+							TournamentPlayerSerializer, \
+							TournamentInProgressSerializer
+from .managers import TournamentSetupManager, \
+						TournamentInProgressManager
+from .exceptions import TournamentSetupException, \
+							TournamentInProgressException
+from .tournamentState import TournamentState
+from User.models import User
+from shared_utilities.decorators import must_be_authenticated, \
+											must_not_be_username, \
+											check_that_valid_tournament_request
+
+import sys
+
+# Create your views here.
+class TournamentListView(APIView):
+	@method_decorator(must_be_authenticated)
+	def post(self, request):
+		body_data = request.data.copy()
+		body_data['host_user'] = request.user.id
+
+		tournament_creation_serializer = TournamentCreationSerializer(data=body_data)
+		if not tournament_creation_serializer.is_valid():
+			return Response({'error': 'invalid request for tournament creation'}, status=status.HTTP_400_BAD_REQUEST)
+		
+		try:
+			tournament = TournamentSetupManager.create_tournament_and_tournament_player_for_host(tournament_creation_serializer.validated_data)
+			host_tournament_player_serializer = TournamentPlayerSerializer(tournament.players.all().first())
+			return Response({
+				'tournament_id': tournament.id,
+				'tournament_player': host_tournament_player_serializer.data
+			}, status=status.HTTP_201_CREATED) 
+		except Exception as e:
+			return Response({'error': str(e)},status=status.HTTP_403_FORBIDDEN)
+		
+class TournamentDetailView(APIView):
+	def get(self, request, tournament_id):
+		tournament = Tournament.objects.filter(id=tournament_id).first()
+		if not tournament:
+			return Response(f"Tournament {tournament_id} not found.", status=status.HTTP_404_NOT_FOUND)
+		serializer = TournamentSerializer(tournament)
+		return Response(serializer.data, status=status.HTTP_200_OK)
+
+	@method_decorator(must_be_authenticated)
+	@method_decorator(check_that_valid_tournament_request([TournamentState.LOBBY, TournamentState.IN_PROGRESS]))
+	def post(self, request, tournament_id):
+		'''If tournament is in lobby, start tournament.
+
+		If tournament is in progress, return next match url.
+		'''
+		tournament = Tournament.objects.get(id=tournament_id)
+		if tournament.state == TournamentState.LOBBY:
+			try:
+				TournamentSetupManager.start_tournament(tournament)
+				serializer = TournamentInProgressSerializer(tournament)
+				return Response(serializer.data, status=status.HTTP_200_OK)
+			except Exception as e:
+				return Response({'error': str(e)}, status=status.HTTP_403_FORBIDDEN)
+
+		elif tournament.state == TournamentState.IN_PROGRESS:
+			pass
+
+	@method_decorator(must_be_authenticated)
+	@method_decorator(check_that_valid_tournament_request([TournamentState.LOBBY, TournamentState.IN_PROGRESS]))
+	def delete(self, request, tournament_id):
+		tournament = Tournament.objects.filter(id=tournament_id).first()
```

Comment:
I dont think we need a delete route, or that the delete route should actually even delete anything but maybe just expire the tournament? But dont know for sure.